lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -942,7 +942,7 @@ def get_mgmt_svc_client(self, client_type, base_url=None, api_version=None):
         # If the client doesn't accept api_version, it's unversioned.
         # If it does, favor explicitly-specified api_version, fall back to api_profile
         if 'api_version' in client_argspec.args:
-            profile_default_version = api_profile_dict.get('default_api_version', None)
+            profile_default_version = api_profile_dict.get('default_api_version', None) if api_profile_dict else None
```

Comment:
is the default  profile_default_version still necessary?

Reply:
why not just remove it, use api_version or not.

Reply:
i wonder what should take precedence, if profile is there, and module specifies api_version, we should maybe set profile to none